frontend/packages/ceph-storage-plugin/src/components/volume-snapshot/volume-snapshot.tsx

```diff
@@ -0,0 +1,233 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import * as classNames from 'classnames';
+
+import {
+  BreadCrumbs,
+  Kebab,
+  ResourceIcon,
+  ResourceKebab,
+  ResourceLink,
+  SectionHeading,
+} from '@console/internal/components/utils';
+import { ListPage, Table, TableData, TableRow } from '@console/internal/components/factory';
+import { k8sGet, referenceFor } from '@console/internal/module/k8s';
+
+import { Link } from 'react-router-dom';
+import { PersistentVolumeClaimModel } from '@console/internal/models';
+import { VolumeSnapshotModel } from '../../models';
+import { getKebabActionsForKind } from '../../utils/kebab-actions';
+import { sortable } from '@patternfly/react-table';
+import { volumeSnapshotModal } from '../modals/volume-snapshot-modal/volume-snapshot-modal';
+
+const snapshotMenuActions = [...getKebabActionsForKind(VolumeSnapshotModel)];
+
+const snapshotTableColumnClasses = [
+  '',
+  classNames('pf-m-hidden', 'pf-m-visible-on-sm', 'pf-u-w-16-on-lg'),
+  classNames('pf-m-hidden', 'pf-m-visible-on-lg'),
+  classNames('pf-m-hidden', 'pf-m-visible-on-xl'),
+  classNames('pf-m-hidden', 'pf-m-visible-on-2xl'),
+  Kebab.columnClass,
+];
+
+const VolumeSnapshotTableHeader = () => {
+  return [
+    {
+      title: 'Name',
+      sortField: 'metadata.name',
+      transforms: [sortable],
+      props: { className: snapshotTableColumnClasses[0] },
+    },
+    {
+      title: 'Date',
+      sortField: 'metadata.creationTimestamp',
+      transforms: [sortable],
+      props: { className: snapshotTableColumnClasses[1] },
+    },
+    {
+      title: 'Status',
+      sortField: 'status.phase',
+      transforms: [sortable],
+      props: { className: snapshotTableColumnClasses[2] },
+    },
+    {
+      title: 'Size',
+      sortField: 'status.capacity.storage',
```

Comment:
Won't this sort alphabetically? You need a special helper that converts to the base value.